examples/sklearn/demo_grid_search_reduction_regression_sklearn.ipynb

```diff
@@ -1,718 +1,580 @@
+      "Requirement already satisfied: certifi>=2017.4.17 in /Users/ektabhaskar/miniconda3/envs/aif360/lib/python3.11/site-packages (from requests->tempeh->aif360[LawSchoolGPA,Reductions]) (2023.7.22)\n",
+      "Requirement already satisfied: tqdm>=4.27.0 in /Users/ektabhaskar/miniconda3/envs/aif360/lib/python3.11/site-packages (from shap->tempeh->aif360[LawSchoolGPA,Reductions]) (4.66.1)\n",
+      "Requirement already satisfied: slicer==0.0.7 in /Users/ektabhaskar/miniconda3/envs/aif360/lib/python3.11/site-packages (from shap->tempeh->aif360[LawSchoolGPA,Reductions]) (0.0.7)\n",
+      "Requirement already satisfied: numba in /Users/ektabhaskar/miniconda3/envs/aif360/lib/python3.11/site-packages (from shap->tempeh->aif360[LawSchoolGPA,Reductions]) (0.58.0)\n",
+      "Requirement already satisfied: cloudpickle in /Users/ektabhaskar/miniconda3/envs/aif360/lib/python3.11/site-packages (from shap->tempeh->aif360[LawSchoolGPA,Reductions]) (2.2.1)\n",
+      "Requirement already satisfied: llvmlite<0.42,>=0.41.0dev0 in /Users/ektabhaskar/miniconda3/envs/aif360/lib/python3.11/site-packages (from numba->shap->tempeh->aif360[LawSchoolGPA,Reductions]) (0.41.0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "#Install aif360\n",
+    "#Install Reductions from Fairlearn\n",
+    "#Install Low School GPA dataset\n",
+    "!pip install 'aif360[Reductions,LawSchoolGPA]'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "id": "sivw3vma71DE"
+   },
+   "outputs": [
     {
-      "cell_type": "markdown",
-      "metadata": {
-        "id": "lukGdCjv71DJ"
-      },
-      "source": [
-        "### Running metrics"
-      ]
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:root:No module named 'tensorflow': AdversarialDebiasing will be unavailable. To install, run:\n",
+      "pip install 'aif360[AdversarialDebiasing]'\n"
+     ]
     },
     {
-      "cell_type": "markdown",
-      "metadata": {
-        "id": "x9LheHu-71DK"
-      },
-      "source": [
-        "With the data in this format, we can easily train a scikit-learn model and get predictions for the test data. We drop the protective attribule columns so that they are not used in the model."
-      ]
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/Users/ektabhaskar/Desktop/GHC/AIF360/examples/sklearn/demo_grid_search_reduction_regression_sklearn.ipynb\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import sys\n",
+    "import os\n",
+    "\n",
+    "from sklearn.compose import TransformedTargetRegressor\n",
+    "from sklearn.linear_model import LinearRegression\n",
+    "from sklearn.metrics import mean_absolute_error\n",
+    "from sklearn.preprocessing import MinMaxScaler\n",
+    "current_script_path=os.path.abspath('demo_grid_search_reduction_regression_sklearn.ipynb')\n",
+    "print(current_script_path)\n",
```

Comment:
Perhaps we'd like to clean this part up for review 👍 

Also, it looks like there are a few warnings noticed in https://github.com/EktaBhaskar/AIF360/blob/master/examples/sklearn/demo_grid_search_reduction_regression_sklearn.ipynb, do you also see those warnings locally as well?

Reply:
@yehee I'm not sure why but the notebook was giving error to me on simply importing like:-

`from aif360.sklearn.datasets.seaphe_datasets import load_lawschool_data
from aif360.sklearn.inprocessing import GridSearchReduction
from aif360.sklearn.metrics import difference`

I assumed it might be because of some dependancy issues.
So I had to include sys.path.
Do you want me to write import statements as above? It might work on proper setup, just won't work on mine.

Yes, I can see warnings but not sure why it's there as I did not make any changes related to AdversarialDebiasing.

Reply:
if you replace the line in [`__init__.py`](https://github.com/Trusted-AI/AIF360/blob/master/aif360/sklearn/datasets/__init__.py#L15) with the `seaphe_datasets` file, it should import normally. The warning is fine but there's also an error later. It looks like you commented out
```py
from aif360.sklearn.metrics import difference
```

Reply:
I have tried putting in __init__.py of sklearn folder as well. But it gives me error:-

this is __init__.py:-

```

from aif360.sklearn.datasets.utils import standardize_dataset, NumericConversionWarning
from aif360.sklearn.datasets.openml_datasets import fetch_adult, fetch_german, fetch_bank
from aif360.sklearn.datasets.compas_dataset import fetch_compas
from aif360.sklearn.datasets.meps_datasets import fetch_meps
from aif360.sklearn.datasets.tempeh_datasets import fetch_lawschool_gpa
from aif360.sklearn.datasets.seaphe_datasets import load_lawschool_data

```

error:-

```
ImportError: cannot import name 'load_lawschool_data' from 'aif360.sklearn.datasets' (/opt/homebrew/lib/python3.10/site-packages/aif360/sklearn/datasets/__init__.py)

```

Reply:
I had to comment :-
```from aif360.sklearn.metrics import difference```

Because it was giving me error even before I made any changes and was just trying to run this notebook to see how it works. And later on also, did not make any changes in this method/class:-

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In [28], line 14
     12 from aif360.sklearn.inprocessing import GridSearchReduction
     13 # from aif360.sklearn.datasets import load_lawschool_data
---> 14 from aif360.sklearn.metrics import difference

File /opt/homebrew/lib/python3.10/site-packages/aif360/sklearn/metrics/__init__.py:7
      1 """
      2 ``aif360.sklearn`` implements a number of fairness metrics for group fairness
      3 and individual fairness. For guidance on which metric to use for a given
      4 application, see our
      5 `Guidance <http://aif360.mybluemix.net/resources#guidance>`_ page.
      6 """
----> 7 from aif360.sklearn.metrics.metrics import *

File /opt/homebrew/lib/python3.10/site-packages/aif360/sklearn/metrics/metrics.py:14
     11 from sklearn.utils.deprecation import deprecated
     13 from aif360.sklearn.utils import check_inputs, check_groups
---> 14 from aif360.detectors.mdss.ScoringFunctions import BerkJones, Bernoulli
     15 from aif360.detectors.mdss.MDSS import MDSS
     17 __all__ = [
     18     # meta-metrics
     19     'difference', 'ratio', 'intersection', 'one_vs_rest',
   (...)
     38     'false_positive_rate_error'
     39 ]

File /opt/homebrew/lib/python3.10/site-packages/aif360/detectors/mdss/ScoringFunctions/__init__.py:5
      3 from aif360.detectors.mdss.ScoringFunctions.Poisson import Poisson
      4 from aif360.detectors.mdss.ScoringFunctions.BerkJones import BerkJones
----> 5 from aif360.detectors.mdss.ScoringFunctions.Gaussian import Gaussian

File /opt/homebrew/lib/python3.10/site-packages/aif360/detectors/mdss/ScoringFunctions/Gaussian.py:1
----> 1 from turtle import pen
      2 from aif360.detectors.mdss.ScoringFunctions.ScoringFunction import ScoringFunction
      3 from aif360.detectors.mdss.ScoringFunctions import optim

File /opt/homebrew/Cellar/python@3.10/3.10.10_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/turtle.py:107
    103 _ver = "turtle 1.1b- - for Python 3.1   -  4. 5. 2009"
    105 # print(_ver)
--> 107 import tkinter as TK
    108 import types
    109 import math

File /opt/homebrew/Cellar/python@3.10/3.10.10_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/tkinter/__init__.py:37
     34 import sys
     35 import types
---> 37 import _tkinter # If this fails your Python may not be configured for Tk
     38 TclError = _tkinter.TclError
     39 from tkinter.constants import *

ModuleNotFoundError: No module named '_tkinter'


```

Reply:
you need to install the code with `pip install --editable .` from the project root